### PR TITLE
Adds the spell Arcane Barrage to wizards

### PR DESCRIPTION
**Pull request by lzimann**
:cl: Lexorion & Lzimann
tweak: Wizards have developed a new spell. It's called Arcane Barrage and it has been reported that it has a similar function to Lesser Summon Guns!
/:cl:

sprites by Lexorion:

barrage

Only thing that changes from Lesser Summon Guns is the sprite and damage type(from brute to burn). 

This PR also adds a way to disallow having certains spells together(only changed to lesser summon guns and arcane barrage so far).

### CODE REVIEW

File: code/modules/projectiles/projectile/magic.dm [hunk 1]
```diff
@@ -374,3 +374,12 @@
...
 	dismemberment = 50
 	nodamage = 0
+
+/obj/item/projectile/magic/arcane_barrage
+	name = "arcane bolt"
+	icon_state = "arcane_barrage"
+	damage = 20
+	damage_type = BURN
+	nodamage = 0
+	armour_penetration = 100
```
Comment (lzimann) [inline code/modules/projectiles/projectile/magic.dm]: I'm just not sure about this value. I don't know if it's too strong. Would like suggestions for it.

File: code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 2]
```diff
@@ -132,12 +132,26 @@
 	name = "enchanted bolt action rifle"
 	desc = "Careful not to lose your head."
 	var/guns_left = 30
+	var/gun_type
 	mag_type = /obj/item/ammo_box/magazine/internal/boltaction/enchanted
 
+/obj/item/weapon/gun/ballistic/shotgun/boltaction/enchanted/arcane_barrage
+	name = "arcane barrage"
+	desc = "Need a desc"
```
Comment (PKPenguin321) [inline code/modules/projectiles/guns/ballistic/shotgun.dm]: Dont forget to fix this

File: code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 3]
```diff
@@ -146,15 +160,15 @@
...
+		GUN.guns_left = guns_left - 1
 		user.drop_item()
 		user.swap_hand()
 		user.put_in_hands(GUN)
 	else
 		user.drop_item()
-	src.throw_at_fast(pick(oview(7,get_turf(user))),1,1)
-	user.visible_message("<span class='warning'>[user] tosses aside the spent rifle!</span>")
+	//throw_at_fast(pick(oview(7,get_turf(user))),1,1)
+	//user.visible_message("<span class='warning'>[user] tosses aside the spent rifle!</span>")
```
Comment (PKPenguin321) [inline code/modules/projectiles/guns/ballistic/shotgun.dm]: >commented code

**Comment by KorPhaeron** [comment]
Should just keep both in the spellbook since you kept both spells in the code

**Comment by ChangelingRain** [comment]
I think having both might be neat, maybe. Pick your own damage type.

**Comment by KorPhaeron** [comment]
Looks cool though

**Comment by Kevak** [comment]
I prefer it as a brute damage dealer. Bleeding is p valuable.

**Comment by PKPenguin321** [comment]
Having both might be sketch since you could take both and have effectively a halved cooldown but yeah this is neat

File: code/modules/projectiles/guns/ballistic/shotgun.dm (hunk 3 above)
Comment (Xhuis) [inline code/modules/projectiles/guns/ballistic/shotgun.dm]: You forgot the backslash before the greentext arrow
\>like this

File: code/modules/projectiles/projectile/magic.dm (hunk 1 above)
Comment (Xhuis) [inline code/modules/projectiles/projectile/magic.dm]: Does it really need armor pen at all? 20 burn is very good on its own, and armor types don't have that high of a resistance to it iirc

File: code/modules/projectiles/guns/ballistic/shotgun.dm (hunk 2 above)
Comment (lzimann) [inline code/modules/projectiles/guns/ballistic/shotgun.dm]: well recommend me a desc then

**Comment by lzimann** [comment]
Would be nice having both, but might be waaaay too strong being able to grab both spells at the same time(maybe I can add a check so you can't?)

File: code/modules/projectiles/projectile/magic.dm (hunk 1 above)
Comment (lzimann) [inline code/modules/projectiles/projectile/magic.dm]: I don't really know how armor pen works/how effective lower numbers are.

**Comment by XDTM** [comment]
I like this, but if y'all keep changing wizard i'll never be able to finish wizard mods :'(

File: code/modules/projectiles/guns/ballistic/shotgun.dm (hunk 2 above)
Comment (RemieRichards) [inline code/modules/projectiles/guns/ballistic/shotgun.dm]: pew pew pew

File: code/modules/projectiles/projectile/magic.dm (hunk 1 above)
Comment (RemieRichards) [inline code/modules/projectiles/projectile/magic.dm]: @lzimann it's % of armour ignored.
Armour is in percentages, so 60 Melee, you take 40% of incoming damage.
Armour Penetration is subtracted from Armour before every calculation, but can only take Armour as low as 0 (so you'll never GAIN bonus damage from Armour Penetration), so for 100% penetration you will deal 100% of the dame 100% of the time, so people will take 20 damage, regardless of if they're naked or in full riot gear.

File: code/modules/projectiles/guns/ballistic/shotgun.dm (hunk 2 above)
Comment (Xhuis) [inline code/modules/projectiles/guns/ballistic/shotgun.dm]: Are you looking for serious or goofy descriptions?

Comment (lzimann) [inline code/modules/projectiles/guns/ballistic/shotgun.dm]: Whatever honestly, goofy kinda fits more for wizards' stuff overall though.

**Comment by TechnoAlchemisto** [comment]
Lesser summon guns looks so cool tho

**Comment by lzimann** [comment]
I'll keep both, working on a way so you can't have both spells at the same time right now.

**Comment by KorPhaeron** [comment]
@TechnoAlchemisto 

Why are you such a weaboo

File: code/modules/projectiles/guns/ballistic/shotgun.dm (hunk 2 above)
Comment (Dorsisdwarf) [inline code/modules/projectiles/guns/ballistic/shotgun.dm]: "When a magic missile is not enough"

**Comment by TechnoAlchemisto** [comment]
S-stop

**Comment by kevinz000** [comment]
nice

File: code/modules/spells/spell_types/infinite_guns.dm [hunk 4]
```diff
@@ -10,14 +10,21 @@
...
-		var/obj/item/weapon/gun/ballistic/shotgun/boltaction/enchanted/GUN = new
+		var/GUN = new summon_path
 		C.put_in_hands(GUN)
 		C.swap_hand(C.get_held_index_of_item(GUN))
+
+/obj/effect/proc_holder/spell/targeted/infinite_guns/gun
+
+/obj/effect/proc_holder/spell/targeted/infinite_guns/arcane_barrage
+	name = "Arcane Barrage"
+	desc = "Requires both hands free to use."
```
Comment (lzimann) [inline code/modules/spells/spell_types/infinite_guns.dm]: I probably need a better desc than this but I'm horrible with words and shit

Comment (Lexorion) [inline code/modules/spells/spell_types/infinite_guns.dm]: Something like "Fire a torrent of arcane enrgey at your foes with this (powerful) spell"  perhaps.

**Comment by Supermichael777** [comment]
lazer summon guns

**Comment by YakumoChen** [comment]
except they're both shit compared to Mutate...

**Comment by coiax** [comment]
Isn't the point of lesser summon guns is to have a bunch of spent one use rifles lying around all over the place? If there's no debris, then it's just not as good.

**Comment by Supermichael777** [comment]
the point of it is to be an anime fag

**Comment by lordpidey** [comment]
This spell might actually be better if it's subclassed as a subtype of /obj/effect/proc_holder/spell/targeted/conjure_item/spellpacket

Just have it summon a different item.

**Pull request merged**